sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2127,6 +2127,15 @@ class Dataset[T] private[sql](
       valueColumnName: String): DataFrame =
     unpivot(ids, Array.empty, variableColumnName, valueColumnName)
 
+  /**
+   * Called from Python as Seq[Column] are easier to create via py4j than Array[Column].
+   */
+  private[sql] def _unpivot(ids: Seq[Column],
```

Comment:
I guess we can move it to `src/main/scala/org/apache/spark/sql/api/python/PythonSQLUtils.scala`

Reply:
I'd prefer to get `PythonSQLUtils.toArray` return array of `Column`s but it returns array of `Object`. Looks like generic types do not work as expected.

Reply:
maybe we can directly change the types of `ids` and `values` from `Array[Column]` to `Seq[Column]`?

Reply:
The `Array[Column]` was chosen over `Seq[Column]` to make `Dataset.unpivot` Java-friendly.

Reply:
got it.

since this method is still needed and dedicated only for python, what about move it to `PythonSQLUtils`?

something like

```
def unpivot(dataframe: DataFrame, ids: Seq[Column] ...) = {
    dataframe.unpivot(ids.toArray, values.toArray, variableColumnName, valueColumnName)
}

```

Reply:
Sure, just hesitating because that looks like a very specific method for such a generic utils class. The other methods are more generic. And there are other private Python methods in `Dataset` (`mapInPandas`) and `RelationalGroupedDataset` (`flatMapGroupsInPandas`).

Reply:
can you rename the method? we do not use `_xxx` like function names in scala

Reply:
> And there are other private Python methods in Dataset (mapInPandas) and RelationalGroupedDataset (flatMapGroupsInPandas).

they are also used in test

Reply:
Renamed `private[sql] def _unpivot` to `private[sql] def unpivot`.